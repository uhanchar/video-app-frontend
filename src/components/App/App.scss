@import "src/constants";

.app {
  width: 100%;
  height: 100vh;
  background: #f9f9f9;

  > .container {
    height: calc(100vh - #{ $header-height });
    padding: 0;
  }
}
