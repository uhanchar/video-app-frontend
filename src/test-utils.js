import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const reducerInitialState = {};

function render(
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';

export { render };
