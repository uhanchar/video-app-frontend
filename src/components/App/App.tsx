import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { Container } from '@material-ui/core';

import 'components/App/App.scss';
import Header from 'components/Header/Header';
import VideoList from 'components/VideoList/VideoList';
import UploadVideo from 'components/UploadVideo/UploadVideo';
import Video from 'components/Video/Video';
import { LocationPath } from 'constants/location-path.enum';

const App = () => (
  <div className="app">
    <Header />

    <Container maxWidth="xl" className="container">
      <Switch>
        <Route path={LocationPath.Home} exact component={VideoList} />

        <Route path={LocationPath.Upload} exact component={UploadVideo} />

        <Route path={`${ LocationPath.Video }/:id`} component={Video} />
      </Switch>
    </Container>
  </div>
);

export default App;
