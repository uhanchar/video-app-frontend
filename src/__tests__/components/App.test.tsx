import React from 'react';
import { render as renderWithRedux } from 'test-utils';

import App from 'components/App/App';

describe('test for App component', () => {
  it('should render App component', () => {
    const { asFragment } = renderWithRedux(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
