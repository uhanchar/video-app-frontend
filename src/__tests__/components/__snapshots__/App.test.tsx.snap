// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for App component should render App component 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="header"
      data-testid="header"
    >
      <a
        aria-current="page"
        class="title active"
        href="/"
      >
        Video app
      </a>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        data-testid="header-navigate-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Upload video
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </header>
    <div
      class="MuiContainer-root container MuiContainer-maxWidthXl"
    >
      <div
        class="video-list"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
